### ISSUE TITLE
TextField Accessibility ID not persist in Android but exist in iOS

### ISSUE THREAD

**Issue by Dabbit-Chan** [flutter/flutter#160014]
### Steps to reproduce

1. build a sample app with one `TextField` and one `TextButton`
2. runApp
3. use `appium` to check Accessibility ID exist or not

### Expected results

TextField and TextButton have Accessibility ID in both Android and iOS package

### Actual results

TextField and TextButton have Accessibility ID in iOS package, TextButton have Accessibility ID in Android package and TextField not persist it.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Semantics(
              identifier: 'usernameIdentifier',
              child: const TextField(
                decoration: InputDecoration(
                  hintText: 'Username',
                ),
              ),
            ),
            Semantics(
              identifier: 'loginButtonIdentifier',
              child: TextButton(
                onPressed: () {
                  // todo:
                },
                child: const Text('Login'),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

### iOS(with Accessibility ID)
<img width="1534" alt="zm20241210-1118-2" src="https://github.com/user-attachments/assets/42137993-58f6-406c-91f7-332002942dfe">
<img width="1527" alt="zm20241210-1119-2" src="https://github.com/user-attachments/assets/c77e421b-d6ac-4368-81aa-0883fa376d7d">

### Android(no Accessibility ID)
![18740351-5ECF-426C-A34B-1413AF0EE970](https://github.com/user-attachments/assets/eaffd2a9-1b63-492a-a99f-11eb9cee77b4)
![99535024-891E-43B2-A749-83395BE15262](https://github.com/user-attachments/assets/f07eb7c2-4b0a-497b-92cb-255af1118f20)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.3, on macOS 14.5 23F79 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.90.2)
[✓] Connected device (2 available)
[!] Network resources
    ✗ A cryptographic error occurred while checking "https://maven.google.com/": Handshake error in client
      You may be experiencing a man-in-the-middle attack, your network may be compromised, or you may have malware installed on your computer.

! Doctor found issues in 1 category.

```

</details>

**Comment by darshankawar** [comment]
Can you try passing `label` as well like mentioned [here](https://stackoverflow.com/a/77972933) to see if it helps in your case or not ?

**Comment by Dabbit-Chan** [comment]
I found a similar issue, but nothing valid. https://github.com/flutter/flutter/issues/144357

**Comment by Dabbit-Chan** [comment]
Add label doesn't work, I actually used both label and identifier in my project, but neither worked.

**Comment by darshankawar** [comment]
Thanks for the update. Yes I did see the linked issue which mentions to use `Semantics.identifier` which you are already using. Can you upgrade to latest stable and re-run your scenario to check if it still persist or not ?

**Comment by Dabbit-Chan** [comment]
After upgrade to latest stable(flutter 3.24.5 dart 3.5.4), I still the same thing.

### iOS
<img width="1637" alt="zm20241211-1157-2" src="https://github.com/user-attachments/assets/86477fe9-3130-4149-89dc-d5e217dc1cbc">
<img width="1668" alt="zm20241211-1158-2" src="https://github.com/user-attachments/assets/54a2d7fc-3b87-489d-9874-76b49da7bc16">

### Android
![A91305C7-21A9-483C-8FD6-C620FABCFF28](https://github.com/user-attachments/assets/0454c1e6-28d2-47b5-96fb-555c53747391)
![3525837E-C485-41E9-9EBA-11BEE2EB294B](https://github.com/user-attachments/assets/2c7aeece-7ed5-416f-8200-8975cbc6c7e6)

**Comment by darshankawar** [comment]
Thanks for the update. Seeing the same behavior as reported and seems specific to android.
Original issue that was closed due to inactivity. https://github.com/flutter/flutter/issues/144357

```
stable : 3.24.5
master : 3.27.0-1.0.pre.778
```

**Comment by chunhtai** [comment]
This seems like a appium bug to me, flutter uses setViewIdResourceName to set the id. it would be appium to display it correctly

**Comment by Dabbit-Chan** [comment]
But why this issue only happen in Android package's TextField? I think this is a bug for TextField not appium. @chunhtai

**Comment by chunhtai** [comment]
I am not sure what Android package you are referring to. but it looks like it is appium's android inspector treated resource name of textfield differently than other view
